### ISSUE TITLE
Add Yoda Text Command

### ISSUE THREAD

**Issue by aaravm4** [python-discord/seasonalbot#494]
## Description

I was thinking of adding a `.yoda <text>` command that transforms text into yoda-like speech. 
Example:
`.yoda You must have patience my young Padawan.`
Returns:
`Patience you must have my young Padawan. Hrmm`

## Reasoning

I recently saw the `.snake snakify` command, and thought it would be nice to implement a command that converts normal text to yoda-like text. It seems like a really fun implementation.

## Proposed Implementation

Yoda's speech is always switched from the form of:
`<subject> <action>` instead of `<action> <subject>`

A proposed implementation of switching a sentence would be to iterate through the sentence and finding a verb. Then append that portion from start to the location of that verb to the end. This portion to append at the end should not contain any subjects.

At the end of the text I could add a sound he makes like `Hrmm`.

## Would you like to implement this yourself?

- [x] I'd like to implement this feature myself
- [ ] Anyone can implement this feature

**Comment by gustavwilliam** [comment]
How are you planning to switch the subject and action? Would that require some basic natural language processing?

Also, I don’t think Yoda only adds “hhmmm” at the end. I think randomly putting it in the sentence (between statements) or at the end or beginning would be the way to go.

**Comment by aaravm4** [comment]
I agree that yoda does add other sounds including hhmmm and yes, which I can sprinkle in throughout the text. In terms of switching the subject, verb, and action I could use a nlp parser such as spaCy to accomplish finding the subject. I'll look over some of his lines to get a better understanding of his talk as well.

**Comment by gustavwilliam** [comment]
Alright. How would many sentences and more complex ones be handled? Like the one I just wrote (or this one).

**Comment by aaravm4** [comment]
Let me test around with spaCy and yoda translations, and get back on a more concrete approach.

**Comment by gustavwilliam** [comment]
@aaravm4 How did it go? Any further thoughts on the implementation?

**Comment by aaravm4** [comment]
Hi. I have been pretty busy the past few months, but it's finally Thanksgiving break, so I can spend more time on an implementation.

**Comment by gustavwilliam** [comment]
Alright, sounds good! Life happens; There's no hurry.

**Comment by Xithrius** [comment]
@aaravm4 I'd like to see this feature in Sir-Lancebot. Are you still willing to implement this?

**Comment by aaravm4** [comment]
Yea, I can still implement it.

**Comment by Xithrius** [comment]
Alright, cool.

**Comment by aaravm4** [comment]
Almost done.

**Comment by Xithrius** [comment]
@aaravm4 How's it going?

**Comment by agussarsson** [comment]
Hello! We're a group of students from KTH DD2480, and we'd like to take on this issue. As far as I can tell there are no one assigned yet, would it be possible for me to get assigned?

**Comment by Xithrius** [comment]
@agussarsson assigned!
